src/api/client_server/search.rs

```diff
@@ -162,7 +162,7 @@ pub(crate) async fn search_events_route(body: Ruma<search_events::v3::Request>)
 	let next_batch = if results.len() < limit {
 		None
 	} else {
-		Some((skip + limit).to_string())
+		Some((skip.checked_add(limit).unwrap()).to_string())
```

Comment:
Would be better to extract this into a variable to ensure it's the same value here as in the `for` loop above.

Reply:
Can you clarify what do you mean? I'm not understanding what should be checked against what

Reply:
There's a `skip.saturating_add(limit)` in line 121 above, which used to be the same `skip + limit` as this was, I'm not sure if those two values need to be the same.

Reply:
I see now, yes these should be the same. `next_batch` is a token which is `skip + token` and we iterate that N times, followed by converting it to a string. I'll make that a variable. The way this code was formatted made me seem like they were two separate things.